ansible_rulebook/engine.py

```diff
@@ -266,7 +282,14 @@ async def _stop(self):
             pass
 
         await asyncio.wait([self.pa_runner_task, self.action_loop_task])
-        await self.event_log.put(dict(type="Shutdown"))
+        await self.event_log.put(
+            dict(
+                type="Shutdown",
+                message=obj.message,
+                delay=obj.delay,
+                source_plugin=obj.source_plugin,
+            )
```

Comment:
Please change the definition of the Shutdown message in messages.py to include these new fields.    We should use well defined types instead of arbitrary dicts for messages.

Reply:
Convert the Shutdown message back to a dict for sending over the wire.   But this well defined type ensures that we have the correct data in the message to start with.

Reply:
@benthomasson This is being sent to the event log, which has an extra attribute called **type** which doesn't exist in the Shutdown Message. If we have to put a data type for event_log payloads we would have to do that for all the other messages that we are putting in the event_log. If we need to formalize event_log payload we can do that in a separate PR.

Reply:
ok that's fine if it will go in another PR.